src/psyclone/gocean1p0.py

```diff
@@ -1308,11 +1314,38 @@ def gen_ocl(self, parent):
         qlist = symtab.lookup_with_tag("opencl_cmd_queues").name
         flag = symtab.lookup_with_tag("opencl_error").name
 
-        # Then we call clEnqueueNDRangeKernel
-        parent.add(CommentGen(parent, " Launch the kernel"))
-        cnull = "C_NULL_PTR"
+        # Choose the command queue number to which to dispatch this kernel. We
+        # have do deal with possible dependencies to kernels dispatched in
+        # different command queues as the order of execution is not guaranteed.
         queue_number = self._opencl_options['queue_number']
         cmd_queue = qlist + "({0})".format(queue_number)
+        outer_loop = self.parent.parent.parent.parent
+        dependency = outer_loop.backward_dependence()
+
+        # If the dependency is a loop containing a kernel, add a barrier if the
+        # previous kernels were dispatched in a different command queue
+        if dependency and dependency.coded_kernels():
+            for kernel_dep in dependency.coded_kernels():
+                previous_queue = kernel_dep._opencl_options['queue_number']
```

Comment:
Instead of making the property public I disabled the warning for now and add a TODO to make all the opencl options processing part of the transformation in #1134. I hope this is ok.